data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/S3SourceConfig.java

```diff
@@ -104,6 +104,16 @@ boolean isCodecProvidedWhenNeeded() {
         return true;
     }
 
+    @AssertTrue(message = "acknowledgments and delete_s3_objects_on_read must both be set to true when using PREFIX partition mode")
```

Comment:
Minor: we should probably be consistent on the naming. I see both prefix partition and folder partition being used.

Reply:
good catch this was an outdated message where I initially had a `mode` of PREFIX or FULL_OBJECT_KEY